app/controllers/api/v0/articles_controller.rb

```diff
@@ -80,7 +80,7 @@ def set_user
 
       def article_params
         params["article"].transform_keys!(&:underscore)
-        params["article"]["organization_id"] = (@user.organization_id if params["article"]["post_under_org"])
+        params["article"]["organization_id"] = (@article&.organization_id || @article&.user&.organization_id || @user.organization_id if params["article"]["post_under_org"])
```

Comment:
This uses the article's original `org_id` or the article author's `org_id`, or finally the `@user/current_user`'s organization ID, in that order.